lib/Semantics/check-select-stmt.cpp

```diff
@@ -0,0 +1,393 @@
+//===-- lib/semantics/check-select-stmt.cc --------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "check-select-stmt.h"
+#include "flang/Parser/message.h"
+#include "flang/Semantics/tools.h"
+
+namespace Fortran::semantics {
+
+class SelectCaseHelper {
+
+public:
+  SelectCaseHelper(parser::Messages &messages, int kind, TypeCategory category)
+    : kind_{kind}, category_(category), messages_{messages} {}
+
+  template<typename T> std::string GetValueAsString(T val) {
+    if constexpr (std::is_same_v<T, bool>) {
+      return std::string(val ? ".TRUE." : ".FALSE.");
+    } else if constexpr (std::is_same_v<T, std::int64_t>) {
+      return std::to_string(val);
+    } else {
+      return parser::QuoteCharacterLiteral(val);
+    }
+  }
+  template<typename T> std::string GetValuePairAsString(T fromVal, T toVal) {
```

Comment:
Should be `static`.

Reply:
Similar to GetValueAsString, even this function can be made generic and can be moved to formatting.cpp file.